src/services/ui/src/pages/detail/setup/spa.tsx

```diff
@@ -0,0 +1,114 @@
+import { removeUnderscoresAndCapitalize } from "@/utils";
```

Comment:
While I do like the templating approach, I foresee a scenario where each Detail Item will require its own custom view/styles.  

So, Im of the opinion that these should be centralized components instead of templates.  And later, when these views become clearer defined, we can hammer down the nail with templating. 

Also, we can simplify the business logic of auth-cms by consolidating it here, instead of managing it in different places.

```js


```



```js
export const DetailItemsGrid = (props: { label: string; value: ReactNode }) => {
  return (
    <div key={props.label}>
      <h3 className="text-sm">{props.label}</h3>
      {props.value}
    </div>
  );
};

export const SpaDetails = (data: OsMainSourceItem) => {
  const { data: user } = useGetUser();

  return (
    <>
      <div className="grid grid-cols-2 gap-4">
        <DetailItemsGrid label="Submission ID" value={data.id} />
        <DetailItemsGrid label="State" value={data.state} />
        <DetailItemsGrid
          label="Type"
          value={removeUnderscoresAndCapitalize(data.planType)}
        />
        <DetailItemsGrid
          label="Action Type"
          value={
            data.actionType
              ? LABELS[data.actionType as keyof typeof LABELS] ||
                data.actionType
              : BLANK_VALUE
          }
        />
        <DetailItemsGrid
          label="Initial Submission Date"
          value={
            data.submissionDate
              ? format(new Date(data.submissionDate), "MM/dd/yyyy h:mm:ss a")
              : BLANK_VALUE
          }
        />

        <DetailItemsGrid
          label="Proposed Effective Date"
          value={
            data.proposedDate
              ? format(new Date(data.proposedDate), "MM/dd/yyyy")
              : BLANK_VALUE
          }
        />

        <DetailItemsGrid
          label="Approved Effective Date"
          value={
            data.approvedEffectiveDate
              ? format(
                  new Date(data.approvedEffectiveDate),
                  "MM/dd/yyyy h:mm:ss a"
                )
              : BLANK_VALUE
          }
        />

        {user?.isCms && (
          <DetailItemsGrid
            label="Status Date"
            value={
              data.statusDate
                ? format(new Date(data.statusDate), "MM/dd/yyyy h:mm:ss a")
                : BLANK_VALUE
            }
          />
        )}

        <DetailItemsGrid
          label="Final Disposition Date"
          value={
            data.finalDispositionDate
              ? format(
                  new Date(data.finalDispositionDate),
                  "MM/dd/yyyy h:mm:ss a"
                )
              : BLANK_VALUE
          }
        />
      </div>
      <hr className="my-4" />
    </>
  );
};

export const SubmissionDetails = (data: OsMainSourceItem) => {
  const { data: user } = useGetUser();

  return (
    <>
      <div className="grid grid-cols-2 gap-4">
        <DetailItemsGrid
          label="Submitted By"
          value={
            <p className="text-lg">{data?.submitterName || BLANK_VALUE}</p>
          }
        />
        <DetailItemsGrid
          label="Submission Source"
          value={
            <p className="text-lg">
              {data?.origin?.toLowerCase() === "onemac"
                ? "OneMAC"
                : BLANK_VALUE}
            </p>
          }
        />
        {user?.isCms && (
          <DetailItemsGrid
            label="Subject"
            value={<p>{data?.subject || BLANK_VALUE}</p>}
          />
        )}

        {user?.isCms && (
          <DetailItemsGrid
            label="Description"
            value={<p>{data?.description || BLANK_VALUE}</p>}
          />
        )}
        <DetailItemsGrid
          label="CPOC"
          value={
            <p className="text-lg">{data?.leadAnalystName || BLANK_VALUE}</p>
          }
        />

        {user?.isCms && (
          <DetailItemsGrid
            label="Review Team (SRT)"
            value={<ReviewTeamList team={data.reviewTeam} />}
          />
        )}
      </div>
      <hr className="my-4" />
    </>
  );
};
```

But, I'll leave it to you. It's your call.

Reply:
This is honestly just a clean-up job for the pattern already used -- there was a good bit of duplicated code to remove -- but we should definitely do a full consideration refactor of it at some point with a larger establishment of patterns/norms.